cf/rds.yaml

```diff
@@ -0,0 +1,123 @@
+
+# --------------------------------------- Constants ---------------------------------------
+  DBPort:
+    Type: String
+    Default: "5432"
+
+  DBEngineVersion:
+    Type: Number
+    Default: 13.4
+
+  DBSecretName:
+    Description: The name of the secret used to store master user and password.
+    Type: String
+    Default: aurora-staging
+
+Resources:
+  DBCluster:
+    Type: AWS::RDS::DBCluster
+    DeletionPolicy: Snapshot
+    UpdateReplacePolicy: Snapshot
+    Properties:
+      Engine: aurora-postgresql
+      EngineMode: provisioned
+      EngineVersion: !Ref DBEngineVersion
+      DBSubnetGroupName: !Ref DBSubnetGroup
+      DBClusterIdentifier: !Join [ "-", ["aurora-cluster", !Ref Environment] ]
+      DatabaseName: "aurora_db"
+      Port: !Ref DBPort
+      # Can't be deleted before first setting this to false, should be true when we are done with the basic setup of db
+      DeletionProtection: false
+      MasterUsername: !Join ["", ["{{resolve:secretsmanager:", !Ref DBSecretName, ":SecretString:username}}" ]]
+      MasterUserPassword: !Join ["", ["{{resolve:secretsmanager:", !Ref DBSecretName, ":SecretString:password}}" ]]
+      EnableIAMDatabaseAuthentication: true
+      BackupRetentionPeriod: 5
+      SourceRegion: eu-west-1
+      # Check if we want to change this
+      StorageEncrypted: false
+      VpcSecurityGroupIds: 
+        - !Ref RDSSecurityGroup
+
+  DBSubnetGroup:
+    Type: AWS::RDS::DBSubnetGroup
+    Properties:
+      DBSubnetGroupDescription: !Sub Biomage subnet group for ${Environment}.
+      DBSubnetGroupName: !Sub rds-subnet-group-${Environment}
+      SubnetIds:
+        Fn::Split: [',', Fn::ImportValue: !Sub "eksctl-biomage-${Environment}-cluster::SubnetsPrivate"]
+      Tags:
+        - Key: "Name"
+          Value: AuroraSubnetGroup
+
+  DBInstance1:
+    Type: AWS::RDS::DBInstance
+    Properties:
+      DBInstanceClass: db.t3.medium
+      DBClusterIdentifier: !Ref DBCluster
+      Engine: aurora-postgresql
+      EngineVersion: !Ref DBEngineVersion
+      AutoMinorVersionUpgrade: true
+      PubliclyAccessible: false
+  
+  RDSAgentSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: !Sub Biomage RDS security group for ${Environment} to connect ec2 agent with rds
+      VpcId:
+        Fn::ImportValue:
+          !Sub 'eksctl-biomage-${Environment}-cluster::VPC'
+    # this says who can access anyone within this security group
+      SecurityGroupIngress:
+        - FromPort: !Ref DBPort
+          ToPort: !Ref DBPort
+          IpProtocol: tcp
+          CidrIp: 0.0.0.0/0
+
+  RDSSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: !Sub "Biomage RDS security group for ${Environment}"
+      SecurityGroupEgress: []
+      SecurityGroupIngress:
+        - SourceSecurityGroupId: 
+            Fn::ImportValue: !Sub "eksctl-biomage-${Environment}-cluster::SharedNodeSecurityGroup"
```

Comment:
if `SharedNodeSecurityGroup` is not used anymore (from your comment), then why do you use it in this line?

Reply:
It is used in redis alongside the not outdated security, so I would prefer to have the removal be in another ticket, since it involves removing this from redis as well and also going manually and removing the staging and production security groups created from this template.

Reply:
did you create a ticket for this?

Reply:
Makes sense about removing the `SharedNodeSecurityGroup `as part of another ticket. About the cleanup of those staging and production security groups that got created as part of this template - why not doing this now? You created them recently, so they are still fresh in your head. After some time, will you remember what you did?

Reply:
I misspoke, I didn't create the `SharedNodeSecurityGroup` and my templates weren't creating it, they were just referencing it. 

It is created in some other way, I don't know how exactly, maybe just through the aws console, but they are referenced by redis, so I think removing them without removing them from the redis template could cause problems.

Reply:
Thanks for the clarification. Looks like I understood about `SharedNodeSecurityGroup` correctly. My question was about the other security groups that you had created when you were figuring out the right configuration for rds. Did you delete them?

Reply:
oh right, I had deleted those (and cant find any of them now when searching apart from the ones we are using for staging currently).